streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -142,35 +144,61 @@ public void to(String topic) {
 
     @Override
     public void to(String topic, Serializer<K> keySerializer, Serializer<V> valSerializer) {
-        String name = topology.newName(SINK_NAME);
-
-        topology.addSink(name, topic, keySerializer, valSerializer, this.name);
+        this.toStream().to(topic, keySerializer, valSerializer);
     }
 
     @Override
     public KStream<K, V> toStream() {
         String name = topology.newName(TOSTREAM_NAME);
 
-        topology.addProcessor(name, new KStreamPassThrough(), this.name);
+        topology.addProcessor(name, new KStreamMapValues<K, Change<V>, V>(new ValueMapper<Change<V>, V>() {
+            @Override
+            public V apply(Change<V> change) {
+                return change.newValue;
+            }
+        }), this.name);
 
         return new KStreamImpl<>(topology, name, sourceNodes);
     }
 
+    @SuppressWarnings("unchecked")
     KTableValueGetterSupplier<K, V> valueGetterSupplier() {
         if (processorSupplier instanceof KTableSource) {
-            final KTableSource<K, V> source = (KTableSource<K, V>) processorSupplier;
-            synchronized (source) {
-                if (!source.isMaterialized()) {
-                    StateStoreSupplier storeSupplier =
-                        new KTableStoreSupplier<>(source.topic, keySerializer, keyDeserializer, valSerializer, valDeserializer, null);
-                    // mark this state is non internal hence it is read directly from a user topic
-                    topology.addStateStore(storeSupplier, false, name);
-                    source.materialize();
-                }
-            }
+            KTableSource<K, V> source = (KTableSource<K, V>) processorSupplier;
+            materialize(source);
             return new KTableSourceValueGetterSupplier<>(source.topic);
         } else {
-            return processorSupplier.view();
+            return ((KTableProcessorSupplier<K, S, V>) processorSupplier).view();
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    void enableSendingOldValues() {
```

Comment:
Right now we have functions called ```enableSendingOldValues``` in two places, one in KTableImpl and one in KTableProcessorSupplier. Could we remove the one in KTableImpl and only trigger it in KTableProcessorSupplier (btw to do that we can do change KTableSource back to extend from KTableProcessorSupplier)?